pkg/controller/controller.go

```diff
@@ -512,6 +514,42 @@ func unmarshalParameters(in []byte) (map[string]interface{}, error) {
 	return parameters, nil
 }
 
+// fetchSecretParameters requests and returns the contents of the given secret as a map
+func fetchSecretParameters(kubeClient kubernetes.Interface, namespace string, secretRef *v1.LocalObjectReference) (map[string]interface{}, error) {
+	if kubeClient == nil {
+		return nil, errors.New("kubeClient is not defined")
+	}
+	secret, err := kubeClient.CoreV1().Secrets(namespace).Get(secretRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[string]interface{})
+	for k, v := range secret.Data {
+		var parameters interface{}
+		if json.Unmarshal(v, &parameters) != nil {
```

Comment:
>  and you won't see it with raw REST API.

actually you probably will, since there is no way of passing raw byte array in JSON. still, I don't think we need to deal with base64 when we use client-go for working with secrets.